aiida/tools/importexport/common/archive.py

```diff
@@ -40,59 +60,135 @@ class Archive(object):
             archive.version
 
     """
+    MANDATORY_DATA_KEYS = {'node_attributes', 'node_extras', 'export_data', 'links_uuid', 'groups_uuid'}
+    MANDATORY_METADATA_KEYS = {'aiida_version', 'export_version', 'all_fields_info', 'unique_identifiers'}
 
-    FILENAME_DATA = 'data.json'
-    FILENAME_METADATA = 'metadata.json'
-
-    def __init__(self, filepath):
+    def __init__(self, filepath, output_filepath=None, overwrite=False, silent=True, sandbox_in_repo=True):
         self._filepath = filepath
+        self._output_filepath = output_filepath
+        self._overwrite = overwrite
+        self._silent = silent
+        self._sandbox_in_repo = sandbox_in_repo
         self._folder = None
         self._unpacked = False
+        self._keep = False
```

Comment:
Very sensible idea!

However, this is somewhat coupled with the fact that I am still trying to get around having to copy all the files (if they are an extracted archive - a folder on the hdd) before repacking. But maybe there is no getting around this, if we wish to use this class for migrations and the archived node repository files need to be altered. If it was only the JSON files, it may have been possible. Unless the class could handle a selective copying and altering during migrations that the migration functions do not need to know about?